### PR TITLE
[exporter/signalfx] Fix correlation timeout bug

### PR DESCRIPTION
**Pull request by crobert-1**
**Description:** <Describe what has changed.>
The SignalFx Exporter supports correlations between metrics and traces. This means that when a trace occurs from a given host, with dimensions like service and environment, these dimensions can be used to query metrics for these traces. These correlations are supposed to timeout after the stale_service_timeout amount of time passes, but they weren't. This fix makes sure correlations timeout so that metrics aren't linked with stale trace data.

**Testing:**
Manual testing is working as expected, but it's currently impossible (from my understanding) to write tests for this functionality. The problem is that the tests would need to create both metrics and traces to send to the signalfx backend, but at that high of a level the correlation values couldn't be properly checked, as they're not available outside the local package.

### CODE REVIEW

File: internal/coreinternal/timeutils/ticker_helper.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+func (pt *PolicyTicker) OnTick() {
+	pt.OnTickFunc()
+}
+func (pt *PolicyTicker) Stop() {
+	close(pt.StopCh)
+	pt.Ticker.Stop()
+}
+
+// Ensure PolicyTicker implements TTicker interface
+var _ TTicker = (*PolicyTicker)(nil)
```
Comment (pmcollins) [inline internal/coreinternal/timeutils/ticker_helper.go:58]: Can we move this assertion up to right above or below the PolicyTicker struct definition?

**Review by pmcollins — APPROVED** [review]
LGTM, just a couple of nits. Thanks for fixing this!

File: internal/coreinternal/timeutils/ticker_helper.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+			select {
+			case <-pt.Ticker.C:
+				pt.OnTick()
+			case <-pt.StopCh:
+				return
+			}
+		}
+	}()
+}
+func (pt *PolicyTicker) OnTick() {
```
[the remark this answers is not in the document]
Comment (crobert-1) [reply internal/coreinternal/timeutils/ticker_helper.go:55]: Added, thanks!

File: internal/coreinternal/timeutils/ticker_helper.go (hunk 1 above)
Comment (crobert-1) [reply internal/coreinternal/timeutils/ticker_helper.go]: I've moved it right under the struct definition.

**Comment by crobert-1** [comment]
_Failing tests are unrelated to my changes_

**Review by dmitryax — APPROVED** [review]
LGTM

**Review by codeboten — APPROVED** [review]

**Pull request merged**